### PR TITLE
Improve html5 locale fallback selection

### PR DESCRIPTION
**Pull request by KDSBrowne**
Previously, if the browsers locale was set to "ja", the client would default to "en" because it would not find any "ja.json".

Now, it will check for a viable fallback option.

Example : 
localefallbackorder

also adjusts the width of the userlist re-size handle to match that of the chat.

### CODE REVIEW

File: bigbluebutton-html5/imports/startup/server/index.js [hunk 1]
```diff
@@ -21,17 +21,35 @@ WebApp.connectHandlers.use('/check', (req, res) => {
...
-  const defaultLocale = APP_CONFIG.defaultSettings.application.locale;
-  const localeRegion = req.query.locale.split(/[-_]/g);
-  const localeList = [defaultLocale, localeRegion[0]];
-
-  let normalizedLocale = localeRegion[0];
+  const fallback = APP_CONFIG.defaultSettings.application.fallbackLocale;
+  const browserLocale = req.query.locale.split(/[-_]/g);
+
+  const localeList = [fallback];
+  const getAvailableLocales = fs.readdirSync('assets/app/locales');
```
Comment (Tainan404) [inline bigbluebutton-html5/imports/startup/server/index.js]: has the same path being used in two places, why not change to a variable?

File: bigbluebutton-html5/imports/startup/server/index.js [hunk 2]
```diff
@@ -21,17 +21,35 @@ WebApp.connectHandlers.use('/check', (req, res) => {
...
+  let regionDefault = null;
+  const usableLocales = [];
+
+  getAvailableLocales
+    .map(file => file.replace('.json', ''))
+    .map(locale => (
+      locale
+    ))
+    .reduce((i, locale) => {
```
Comment (oswaldoacauan) [inline bigbluebutton-html5/imports/startup/server/index.js]: Hmm, is this correct? With this code you are using a reduce as a for-loop and not as a reducer.

Comment (oswaldoacauan) [reply bigbluebutton-html5/imports/startup/server/index.js]:
```

File: bigbluebutton-html5/imports/startup/server/index.js [hunk 3]
```diff
@@ -21,17 +23,31 @@ WebApp.connectHandlers.use('/check', (req, res) => {
...
-  const defaultLocale = APP_CONFIG.defaultSettings.application.locale;
-  const localeRegion = req.query.locale.split(/[-_]/g);
-  const localeList = [defaultLocale, localeRegion[0]];
+  const fallback = APP_CONFIG.defaultSettings.application.fallbackLocale;
+  const browserLocale = req.query.locale.split(/[-_]/g);
+  const localeList = [fallback];
+
+  const usableLocales = AVAILABLE_LOCALES
+    .map(file => file.replace('.json', ''))
+    .map(locale => (
```
Comment (oswaldoacauan) [inline bigbluebutton-html5/imports/startup/server/index.js]: Do you really need this map?

File: bigbluebutton-html5/imports/startup/server/index.js [hunk 4]
```diff
@@ -21,17 +21,35 @@ WebApp.connectHandlers.use('/check', (req, res) => {
...
+      locale
+    ))
+    .reduce((i, locale) => {
+      if (locale === browserLocale[0]) regionDefault = locale;
+      if (locale.match(browserLocale[0])) usableLocales.push(locale);
+    }, 0);
+
+  if (regionDefault) localeList.push(regionDefault);
+  if (!regionDefault && usableLocales[0]) localeList.push(usableLocales[0]);
```
Comment (Tainan404) [inline bigbluebutton-html5/imports/startup/server/index.js]: `!regionDefault && usableLocales.length`

File: bigbluebutton-html5/imports/ui/components/app/styles.scss [hunk 5]
```diff
@@ -120,7 +121,7 @@ $bars-padding: $lg-padding-x - .45rem; // -.45 so user-list and chat title is al
 
 .userlistPad {
   background-color: $color-off-white;
-  width: $border-size-large;
+  width: $userlist-handle-width;
```
Comment (Tainan404) [inline bigbluebutton-html5/imports/ui/components/app/styles.scss]: Why is the userlist changed here?

File: bigbluebutton-html5/imports/startup/server/index.js [hunk 6]
```diff
@@ -21,17 +23,31 @@ WebApp.connectHandlers.use('/check', (req, res) => {
...
+      locale
+    ))
+    .reduce((locales, locale) =>
+      (locale.match(browserLocale[0]) ? [...locales, locale] : locales), []);
+
+  const regionDefault = usableLocales.find(locale => browserLocale[0] === locale);
 
-  let normalizedLocale = localeRegion[0];
+  if (regionDefault) localeList.push(regionDefault);
+  if (!regionDefault && usableLocales[0]) localeList.push(usableLocales[0]);
```
Comment (Tainan404) [inline bigbluebutton-html5/imports/startup/server/index.js]: `!regionDefault && usableLocales.length`

File: bigbluebutton-html5/imports/ui/components/app/styles.scss (hunk 5 above)
Comment (KDSBrowne) [reply bigbluebutton-html5/imports/ui/components/app/styles.scss]: So the userlist resize handle renders as the same size as the chat resize handle when scroll bars are visible.

File: bigbluebutton-html5/imports/startup/server/index.js [hunk 7]
```diff
@@ -21,17 +23,28 @@ WebApp.connectHandlers.use('/check', (req, res) => {
...
+  const usableLocales = AVAILABLE_LOCALES
+    .map(file => file.replace('.json', ''))
+    .reduce((locales, locale) =>
+      (locale.match(browserLocale[0]) ? [...locales, locale] : locales), []);
+
+  const regionDefault = usableLocales.find(locale => browserLocale[0] === locale);
 
-  let normalizedLocale = localeRegion[0];
+  if (regionDefault) localeList.push(regionDefault);
+  if (!regionDefault && usableLocales.length) localeList.push(usableLocales[0]);
```
Comment (Tainan404) [inline bigbluebutton-html5/imports/startup/server/index.js]: Why check if regionDefault is empty here? on the line above you already do this

**Pull request merged**